src/zenml/prompts/prompt.py

```diff
@@ -0,0 +1,164 @@
+#  Copyright (c) ZenML GmbH 2023. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Simple prompt abstraction for LLMOps workflows in ZenML."""
+
+import re
+from typing import Any, Dict, List
+
+from pydantic import BaseModel, Field
+
+
+class Prompt(BaseModel):
+    """Simple prompt artifact for ZenML pipelines.
+
+    This is a lightweight prompt abstraction designed to work seamlessly
+    with ZenML's artifact system. Complex operations are implemented as
+    pipeline steps rather than built into this class.
+
+    Prompts are versioned artifacts that can be tracked through pipeline
+    runs and compared in the dashboard.
+
+    Examples:
+        # Simple prompt
+        prompt = Prompt(template="Hello {name}!")
+        formatted = prompt.format(name="Alice")
+
+        # Prompt with default variables
+        prompt = Prompt(
+            template="Translate '{text}' to {language}",
+            variables={"language": "French"}
+        )
+        formatted = prompt.format(text="Hello world")
+
+        # System prompt
+        prompt = Prompt(
+            template="You are a helpful assistant.",
+            prompt_type="system"
+        )
+
+        # Versioned prompt for tracking changes
+        prompt = Prompt(
+            template="Answer: {question}",
+            version="2.0.0"  # Track iterations
```

Comment:
do we have `LATEST` etc like we had with the auto-versioning system we have for the model control plane? Or is it just up to the user to remember to change this version number? (i.e. if they change the prompt but they don't change the version number, what happens?)